### PR TITLE
Added kamon integration for metrics

### PR DESCRIPTION
**Pull request by carloscaldas**
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3226)

### CODE REVIEW

File: src/main/scala/beam/sim/BeamHelper.scala [hunk 1]
```diff
@@ -673,7 +676,13 @@ trait BeamHelper extends LazyLogging {
     } mkString " , "
     logger.info(s"Vehicles assigned to households : $vehicleInfo")
 
-    run(beamServices)
+    val executionSpan = Kamon.serverSpanBuilder("beamStarted", "beam").start()
```
Comment (JustinPihony) [inline src/main/scala/beam/sim/BeamHelper.scala:679]: This should only be set up if `isMetricsEnabled` - as above

File: src/main/scala/beam/sim/config/BeamExecutionConfig.scala [hunk 2]
```diff
@@ -2,4 +2,17 @@ package beam.sim.config
...
-case class BeamExecutionConfig(beamConfig: BeamConfig, matsimConfig: MatsimConfig, outputDirectory: String)
+case class BeamExecutionConfig(beamConfig: BeamConfig, matsimConfig: MatsimConfig, outputDirectory: String) {
+  import BeamExecutionConfig._
+
+  val executionId: String = executionIdFromDirectory(outputDirectory)
+
+}
+
+object BeamExecutionConfig {
+
```
Comment (JustinPihony) [inline src/main/scala/beam/sim/config/BeamExecutionConfig.scala:13]: Why is this in an object? I'd default to keeping it internal to the case class until it is needed external

**Review by JustinPihony — CHANGES_REQUESTED** [review]

**Review by JustinPihony — APPROVED** [review]

**Comment by JustinPihony** [comment]
Can this be merged?

**Comment by nikolayilyin** [comment]
Currently we already have a tool to debug akka messages, we decided to not proceed with Kamon for now

**Pull request closed without merging**